plugins/org.locationtech.udig.tools.jgrass/src/org/locationtech/udig/tools/jgrass/utils/OperationUtils.java

```diff
@@ -95,11 +95,16 @@ protected void moveFeatures( final Display display, IProgressMonitor monitor, IL
         SimpleFeatureType toSchema = toLayer.getSchema();
         SimpleFeatureType selectedSchema = selectedLayer.getSchema();
         int compare = DataUtilities.compare(toSchema, selectedSchema);
-        if (compare != 0) {
+        boolean needsReOrder = false;
+        if (compare == -1) {
             showMessage(display,
                     Messages.getString("OperationUtils_warning"), Messages.getString("OperationUtils_sametypeproblem"), //$NON-NLS-1$ //$NON-NLS-2$
                     MSGTYPE.WARNING);
             return;
+        } else if (compare == 1){
```

Comment:
OK, looks good to me. What do you think about a testcase to verify this behavior of [DataUtilities#compare](https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/data/DataUtilities.java#L383) to check compatibility after upgrading GeoTools ;)

Reply:
Indeed the result of this method is essential for correct behavior of copy and move. Therefore, I added a new set of tests (see DataUtilitiesTest class)